### PR TITLE
add download link to existing data for afform File questions

### PR DESCRIPTION
**Pull request by ufundo**
Overview
----------------------------------------
Adds a file download link to existing answers to Afform File fields.

I've split out the EntityFile change into https://github.com/civicrm/civicrm-core/pull/31017 - so this should follow that.

Before
----------------------------------------
At the moment there's no way that I can see to get an uploaded file back out when submitted through an afform?

After
----------------------------------------
There's a download link when existing answers are pre-filled.

### CODE REVIEW

File: ext/afform/core/ang/af/fields/File.html [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <span ng-if="dataProvider.getFieldData()[$ctrl.fieldName].file_name">
+  <a target="_blank" href="/civicrm/file?reset=1&filename={{ dataProvider.getFieldData()[$ctrl.fieldName].uri }}&mime-type={{ dataProvider.getFieldData()[$ctrl.fieldName].mime_type }}"
   <i class="crm-i {{ dataProvider.getFieldData()[$ctrl.fieldName].icon }}"></i>
```
Comment (seamuslee001) [inline ext/afform/core/ang/af/fields/File.html:3]: given your indenting L4 I think this should be indented as well

**Comment by ufundo** [comment]
retest this please?

**Comment by ufundo** [comment]
Looked again at this. 

The case test touched some pretty gnarly code around duplicating custom field data for files at BAO level... 

I've come to the conclusion that that code was a workaround for the fact EntityFile records have been a bit mangled for files uploaded to Custom Fields. 

I do fear there may be other similar workarounds out there in the world, which fixing this could impact. But it feels a lot more like how it *should* be to me.

**Comment by colemanw** [comment]
@ufundo here's a version that only does the afform side of things (and also handles file permissions & adds test coverage): https://github.com/civicrm/civicrm-core/pull/31459

I still agree we ought to fix the weird mismatch in the EntityFile table but I didn't want to hold up the Afform fix for that.

**Comment by ufundo** [comment]
thanks @colemanw

**Pull request closed without merging**